client/src/components/App.js

```diff
@@ -53,9 +53,39 @@ export default class App extends React.Component {
     super(props);
     this.state = {
       auth: checkAuth(),
+      stats: {
+        communities: 0,
+        users: 0,
+        channels: 0,
+        onlineUsers: 0
+      }
```

Comment:
The initialization of the `stats` state should either be `{rooms: 0, users: 0, onlineUsers: 0, totalMessages: 0}` (since `communities` and `channels` fields aren't being used anywhere) or more preferably an empty object `{}` since it gets over-written as is on line 76.